Library/ViewModels/UpdatePledgeViewModel.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import KsApi
+import PassKit
+import Prelude
+import ReactiveSwift
+
+public protocol UpdatePledgeViewModelInputs {
+  func configureWith(project: Project, reward: Reward, refTag: RefTag?)
+  func pledgeAmountDidUpdate(to amount: Double)
+  func shippingRuleSelected(_ shippingRule: ShippingRule)
+  func traitCollectionDidChange()
+  func viewDidLoad()
+}
+
+public protocol UpdatePledgeViewModelOutputs {
+  var configureSummaryViewControllerWithData: Signal<(Project, Double), Never> { get }
+  var configureWithData: Signal<(project: Project, reward: Reward), Never> { get }
+  var shippingLocationViewHidden: Signal<Bool, Never> { get }
+  var confirmationLabelAttributedText: Signal<NSAttributedString, Never> { get }
+}
+
+public protocol UpdatePledgeViewModelType {
+  var inputs: UpdatePledgeViewModelInputs { get }
+  var outputs: UpdatePledgeViewModelOutputs { get }
+}
+
+public class UpdatePledgeViewModel: UpdatePledgeViewModelType, UpdatePledgeViewModelInputs,
```

Comment:
`final`?

Reply:
Man, we're duplicating a lot of functionality here...I think we're missing the red highlighting for invalid input value ... not really sure what's the best way forward here but we'll probably end up duplicating a lot of code :/